### ISSUE TITLE
About the Cityscapes FID

### ISSUE THREAD

**Issue by veroveroxie** [JunlinHan/DCLGAN#7]
Hi, thanks for your pretrained models!

I downloaded it and used the model to generate city images.

But I cannot get the result 49 as reported in your paper.

I'm wondering on which testing set are you performing evaluation?

Here are my testing results:

<testA, fakeA>=55.49 (using testA)
<realA, fakeA> = 55.47 (png format)
<valA, valA->fakeA> = 140. (using the compressed validation set by CUT)

Thanks.

**Comment by JunlinHan** [comment]
Hello,
Thanks for your question.

The test set is the VAL set of original CityScape dataset. It should contain 500 images in total. 
Did you get the dataset with the following processing?

1: Download the Cityscapes dataset from https://cityscapes-dataset.com/
2: Use the script ./datasets/prepare_cityscapes_dataset.py

**Comment by JunlinHan** [comment]
> thanks for your reply. I tried again but I still only same result, i.e., 55
> 
> Here are some examples images I'm getting,
> 
> ![0_B](https://user-images.githubusercontent.com/44474545/154393438-1ec4c607-d273-46ec-bf41-11aab7f3bd9f.png) ![0_A](https://user-images.githubusercontent.com/44474545/154393441-134c24aa-149e-4511-8f4a-5e3dc8141425.png) ![0_A_true](https://user-images.githubusercontent.com/44474545/154393442-80ed071a-4df5-474a-abfa-b1825192739f.png)
> 
> Could you please check the results are consistent with yours?

Oh thanks for let me know.  
I will investigate this issue (health check later, so might reply with a slight delay) ASAP

**Comment by JunlinHan** [comment]
> thanks for your reply. I tried again but I still only same result, i.e., 55
> 
> Here are some examples images I'm getting,
> 
> ![0_B](https://user-images.githubusercontent.com/44474545/154393438-1ec4c607-d273-46ec-bf41-11aab7f3bd9f.png) ![0_A](https://user-images.githubusercontent.com/44474545/154393441-134c24aa-149e-4511-8f4a-5e3dc8141425.png) ![0_A_true](https://user-images.githubusercontent.com/44474545/154393442-80ed071a-4df5-474a-abfa-b1825192739f.png)
> 
> Could you please check the results are consistent with yours?

Hi,
I justed tested the pre-trained model. Your translated results are consistent. Here are my fid results:
![图片](https://user-images.githubusercontent.com/48272064/154405942-7d933b36-4ba5-41a4-ae7f-d43fab63e1f7.png)

My pytorch-fid version is 0.1.1, could you check if your pytorch-fid version is identical to this?

**Comment by veroveroxie** [comment]
Hi, thanks

I was using 0.2.1, I just downgraded to 0.1.1, but still get 55.49.  May I ask do you have any specific arguments to pass in the test command (e.g., possible crop_size change)? I was using 256x256 

"python test.py --dataroot=datasets/cityscapes --name=city".

I can get the same FID result on horse2zebra, but not this one. 

I also tested on FCN, I can also only get 68.8 on PixAcc not 74 as in your paper.

**Comment by JunlinHan** [comment]
> 

Hi,
Oh that's weird.  I didn't change any auguments, it should be the default setting (256 x 256 res). 

Please find a link here (my prediction results with the pre-trained model):  https://drive.google.com/file/d/1VnUBvvQ9KNIHfm_IxWGJ00JtPjssNChX/view?usp=sharing

**Comment by veroveroxie** [comment]
Debugging is really hard! Fortunately, I matched the FID.

**For future reference, use Pillow=5.0.0 to run prepare_dataset!!**!  I was using Pillow>6, which causes the problem. The first file 0_A.png should be 100 kb and the 100_A.png is 110 kb. (My false ones were 83, 91 kb )


Thank you so much, Junlin, you are so nice!


My last question is do you pick the best epoch or use the last epoch in all your experiments?

**Comment by JunlinHan** [comment]
> Debugging is really hard! Fortunately, I matched the FID.
> 
> **For future reference, use Pillow=5.0.0 to run prepare_dataset!!**! I was using Pillow>6, which causes the problem. The first file 0_A.png should be 100 kb and the 100_A.png is 110 kb. (My false ones were 83, 91 kb )
> 
> Thank you so much, Junlin, you are so nice!
> 
> My last question is do you pick the best epoch or use the last epoch in all your experiments?

That's really great!!! Your information/results can be very helpful for future reference. 
My pleasure to help, which is the purpose of open-sourcing.

The results are from the last epoch. For 200 epochs' training, best FID results occues around 100-150 epoch, where horse->zebra translation can acheive ~37 fid for DCLGAN. However, they are just good in terms of FID (some limitations of FID metric), visually not pleasing.

**Comment by veroveroxie** [comment]
Yes, that makes sense.

Thanks a lot.
